### DISCUSSION TITLE
[Data Importer] Load categories from xml

### DISCUSSION THREAD

**Discussion by fashxp** [pimcore/pimcore#9541] [Q&A / Support]
Good afternoon, can I clarify if I'm doing the right thing, at the moment I want to load categories from xml

Example of a string from xml to be loaded:
   category id="90401">Category_NAME/category>
   category id="198118" parentId="90401">Category_NAME1/category>
   category id="198119" parentId="90401">Category_NAME2/category>

![Скриншот 23-06-2021 104726](https://user-images.githubusercontent.com/43819391/123057320-71dff500-d410-11eb-9f7d-6c8151c1685c.png)

The XML itself is loaded the first field (id) is of the type "number", the second field (value) is "Static text", I wanted to clarify how to add the third field (parentId)

![Скриншот 23-06-2021 105321](https://user-images.githubusercontent.com/43819391/123058211-4578a880-d411-11eb-89ad-6db856778e7c.png)

![Скриншот 23-06-2021 103924](https://user-images.githubusercontent.com/43819391/123058273-51fd0100-d411-11eb-92b6-64093f21aedb.png)

maybe I indicated something wrong ?

_Originally posted by @Makskknd in https://github.com/pimcore/data-importer/issues/41#issuecomment-866615889_

**Comment by fashxp** [comment]
@Makskknd since this is a different question in the original issue than posted [here](https://github.com/pimcore/pimcore/discussions/9539), I created a new discussion for that one... 

> I wanted to clarify how to add the third field (parentId)

Building up hierarchies is possible with defining a `Element Location Update` strategy as follows
![image](https://user-images.githubusercontent.com/8792145/123096966-f3934b00-d42f-11eb-8c9f-fbef83095bb5.png)

So, the importer tires to find a parent for each category based on the parentId field from the import file. 

Important in this context is, that the categories in the import file are in the correct order (so that the parent categories are imported first) and the import is executed sequentially (and not in a parallel way as this would mix up the import order) 
![image](https://user-images.githubusercontent.com/8792145/123097408-6a304880-d430-11eb-9837-969a11cb1586.png)

**Comment by Makskknd** [comment]
Thank you for helping me with my problem, when you start importing manually, such an error is issued, you get a queue created

![Скриншот 23-06-2021 155237](https://user-images.githubusercontent.com/43819391/123099536-f2b3e680-d43a-11eb-8ce3-7f75dbd2aca4.jpg)

**Reply by fashxp** [reply]
can you run the command with -vv ... so that we get some stacktrace ... then we should be able to shed some light into this...

**Reply by fashxp** [reply]
looks like a php error which we cannot catch :(
